src/infrastructure/jira/jira-service.ts

```diff
@@ -298,13 +298,14 @@ class JiraService {
 
 			const storedUrl = response.value;
 
-			if (storedUrl === design.url) {
-				await jiraClient.deleteIssueProperty(
-					issueIdOrKey,
-					issuePropertyKeys.ATTACHED_DESIGN_URL,
-					connectInstallation,
-				);
-			}
+			if (!isString(storedUrl)) return;
+			if (!this.areUrlsOfSameDesign(storedUrl, design.url)) return;
```

Comment:
**note:** Compare URLs strictly by `fileKey` and `nodeId` components to fix the issue with the "Jira" widget in Figma being out of sync with Jira.